ddspipe_participants/src/cpp/participant/dynamic_types/DynTypesParticipant.cpp

```diff
@@ -94,212 +88,84 @@ std::shared_ptr<IReader> DynTypesParticipant::create_reader(
-        internal_notify_type_object_(dyn_type);
+        // Get type information
+        const auto type_info = info.type_information.type_information;
+        const auto type_name = info.type_name;
+
+        rtps::CommonParticipant::on_reader_discovery(participant, reason, info, should_be_ignored);
+
+        notify_type_discovered_(type_info, type_name);
     }
 }
 
-void DynTypesParticipant::on_type_information_received(
-        eprosima::fastdds::dds::DomainParticipant* participant,
-        const fastrtps::string_255 /* topic_name */,
-        const fastrtps::string_255 type_name,
-        const fastrtps::types::TypeInformation& type_information)
+void DynTypesParticipant::on_writer_discovery(
+            fastdds::rtps::RTPSParticipant* participant,
+            fastdds::rtps::WriterDiscoveryStatus reason,
+            const fastdds::rtps::PublicationBuiltinTopicData& info,
+            bool& should_be_ignored)
 {
-    std::string type_name_ = type_name.to_string();
-    const TypeIdentifier* type_identifier = nullptr;
-    const TypeObject* type_object = nullptr;
-    DynamicType_ptr dynamic_type(nullptr);
-
-    // Check if complete identifier already present in factory
-    type_identifier = TypeObjectFactory::get_instance()->get_type_identifier(type_name_, true);
-    if (type_identifier)
+    if (info.guid.guidPrefix != participant->getGuid().guidPrefix)
     {
-        type_object = TypeObjectFactory::get_instance()->get_type_object(type_name_, true);
-    }
+        // Get type information
+        const auto type_info = info.type_information.type_information;
+        const auto type_name = info.type_name;
 
-    // If complete not found, try with minimal
-    if (!type_object)
-    {
-        type_identifier = TypeObjectFactory::get_instance()->get_type_identifier(type_name_, false);
-        if (type_identifier)
-        {
-            type_object = TypeObjectFactory::get_instance()->get_type_object(type_name_, false);
-        }
-    }
+        rtps::CommonParticipant::on_writer_discovery(participant, reason, info, should_be_ignored);
 
-    // Build dynamic type if type identifier and object found in factory
-    if (type_identifier && type_object)
-    {
-        dynamic_type = TypeObjectFactory::get_instance()->build_dynamic_type(type_name_, type_identifier, type_object);
+        notify_type_discovered_(type_info, type_name);
     }
+}
 
-    // Request type object through TypeLookup if not present in factory, or if type building failed
-    if (!dynamic_type)
+void DynTypesParticipant::notify_type_discovered_(
+            const fastdds::dds::xtypes::TypeInformation& type_info,
+            const fastcdr::string_255& type_name)
+{
+    const auto type_id = type_info.complete().typeid_with_size().type_id();
+    fastdds::dds::xtypes::TypeObject dyn_type_object;
+    if (fastdds::dds::RETCODE_OK != fastdds::dds::DomainParticipantFactory::get_instance()->type_object_registry().get_type_object(
```

Comment:
Conclusion after testing: it only makes sense (for our purposes) to propagate a DynamicType created from a complete TypeObject, since from a minimal it is not possible to obtain the IDL representation. Hence, I believe it is fine the way it is now.